modules/standard/List.chpl

```diff
@@ -1239,7 +1239,7 @@ module List {
 
       :arg comparator: A comparator used to sort this list.
     */
-    proc ref sort(comparator=Sort.defaultComparator) {
+    proc ref sort(comparator: ?rec=Sort.defaultComparator) {
```

Comment:
I'd slightly worry about the error message when something non-comparator is passed as the argument. `Sort.sort` checks that but the error message may be slightly misleading and/or it may be better to put that check here, too.

Apparently, there is `chpl_check_comparator` in the Sort module to help with that. If you need to use that here because of better error message, maybe it is a good idea to rename that to make it look more user-facing. It can be noted under #5724

Reply:
The warning as is doesn't look too bad:

```
$CHPL_HOME/modules/standard/List.chpl:1242: In function 'sort':
$CHPL_HOME/modules/standard/List.chpl:1257: error: The comparator myBadComparator requires a 'key(a)', 'compare(a, b)', or 'keyPart(a, i)' method  for element type int(64)
Test.chpl:20: Function 'sort' instantiated as: sort(this: list(int(64),false), comparator: myBadComparator)
```